src/rhsm/certificate.py

```diff
@@ -148,35 +125,18 @@ def __init__(self, content=None):
 
     def _update(self, content):
         if content:
-            x509 = X509.load_cert_string(content)
+            x509 = _certificate.load(pem=content)
+            if x509 is None:
+                raise CertificateException("Error loading certificate")
         else:
-            x509 = X509.X509()
+            x509 = _certificate.X509()
         self.__ext = Extensions(x509)
         self.x509 = x509
 
-        self._parse_subject()
+        self.subj = self.x509.get_subject()
         self.serial = self.x509.get_serial_number()
 
-        self.altName = None
-        try:
-            name_ext = self.x509.get_ext('subjectAltName')
-            if name_ext:
-                self.altName = name_ext.get_value()
-        except LookupError:
-            # This may not be defined, seems to only be used for identity
-            # certificates:
-            pass
-
-    def _parse_subject(self):
-        self.subj = {}
-        subject = self.x509.get_subject()
-        subject.nid['UID'] = 458
-        for key, nid in subject.nid.items():
-            entry = subject.get_entries_by_nid(nid)
-            if len(entry):
-                asn1 = entry[0].get_data()
-                self.subj[key] = str(asn1)
-                continue
+        self.altName = x509.get_extension(name='subjectAltName')
```

Comment:
Do we need an `except` clause here since `subjectAltName` is going to be on every certificate?  Or does `x509` just return `None` or an empty string?